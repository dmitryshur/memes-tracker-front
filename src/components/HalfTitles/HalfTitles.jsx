import React from 'react';
import PropTypes from "prop-types";
import './HalfTitles.scss';

const HalfTitles = (props) => {
  const classList = props.chosen ? 'half-titles hidden' : 'half-titles';

  return (
    <div className={classList}>
      <div className="main-title">
        memes<br />tracker
      </div>
      <div className="subtitle">
        Your constant source of fun
      </div>
    </div>
  );
};

HalfTitles.propTypes = {
  chosen: PropTypes.bool
};

HalfTitles.defaultProps = {
  chosen: false
};

export default HalfTitles;
