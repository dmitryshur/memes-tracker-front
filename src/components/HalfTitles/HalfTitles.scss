.half-titles {
  color: #fff;
  left: 50%;
  position: absolute;
  text-align: center;
  top: 10vh;
  transform: translateX(-50%);
  z-index: 1;

  .main-title {
    font-size: 3.8rem;
    font-weight: 900;
    letter-spacing: 0.1rem;
    line-height: 3.6rem;
    text-transform: uppercase;
  }

  .subtitle {
    font-size: 2.0rem;
    font-weight: 700;
    white-space: nowrap;
  }
}
