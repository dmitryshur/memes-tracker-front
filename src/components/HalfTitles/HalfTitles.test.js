import React from 'react';
import { shallow } from 'enzyme';
import HalfTitles from './HalfTitles.jsx';

test('testing HalfTitles component', () => {
  const wrapper = shallow(<HalfTitles />);
  expect(wrapper).toMatchSnapshot();
});
