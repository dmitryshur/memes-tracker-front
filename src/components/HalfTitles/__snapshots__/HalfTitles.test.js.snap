// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HalfTitles component 1`] = `
<div
  className="half-titles"
>
  <div
    className="main-title"
  >
    memes
    <br />
    tracker
  </div>
  <div
    className="subtitle"
  >
    Your constant source of fun
  </div>
</div>
`;
